src/searchspacepool_impl.h

```diff
@@ -121,8 +121,18 @@ BaseSearchSpace<SearchPointType, DataSetType>* BaseSearchSpacePool<SearchPointTy
     // - get the presorted chunk from L(start) to U(end)
     // - remove points in excess (with set_difference) from start to U(Start) and from L(end) to end
     int is, ie;
-    for (is=steps.size()-1; is>=0; is--) if (steps[is] <= start) break;  is = qMax(is, 0);
-    for (ie=0; ie<steps.size(); ie++)    if (steps[ie] >= end) break;    ie = qMin(ie, steps.size()-1);
+    for (is=steps.size()-1; is>=0; is--) {
+        if (steps[is] <= start) {
+            break;
+        }
+        is = qMax(is, 0);
+    }
+    for (ie=0; ie<steps.size(); ie++) {
+        if (steps[ie] >= end) {
+            break;
+        }
+        ie = qMin(ie, steps.size()-1);
+    }
```

Comment:
I think this is wrong - the braces around `for` should only be around the if/break. The qMax/qMin should be outside the for loop. Can someone confirm this?

Reply:
fixed